zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/OpenSearchIncidentUpdateRepository.java

```diff
@@ -0,0 +1,516 @@
+import org.opensearch.client.opensearch._types.query_dsl.Query;
+import org.opensearch.client.opensearch._types.query_dsl.QueryBuilders;
+import org.opensearch.client.opensearch.core.BulkRequest;
+import org.opensearch.client.opensearch.core.ClearScrollRequest;
+import org.opensearch.client.opensearch.core.CountRequest;
+import org.opensearch.client.opensearch.core.SearchRequest;
+import org.opensearch.client.opensearch.core.SearchResponse;
+import org.opensearch.client.opensearch.core.bulk.BulkOperation;
+import org.opensearch.client.opensearch.core.bulk.BulkResponseItem;
+import org.opensearch.client.opensearch.core.bulk.UpdateOperation;
+import org.opensearch.client.opensearch.core.search.Hit;
+import org.opensearch.client.opensearch.core.search.SourceFilter;
+import org.opensearch.client.opensearch.indices.AnalyzeRequest;
+import org.opensearch.client.opensearch.indices.analyze.AnalyzeToken;
+import org.slf4j.Logger;
+
+public final class OpenSearchIncidentUpdateRepository implements IncidentUpdateRepository {
+  private static final int RETRY_COUNT = 3;
+  private static final List<FieldValue> DELETED_OPERATION_STATES =
+      List.of(
+          FieldValue.of(OperationState.SENT.name()),
+          FieldValue.of(OperationState.COMPLETED.name()));
+  private static final Time SCROLL_KEEP_ALIVE = Time.of(t -> t.time("1m"));
+  private static final int SCROLL_PAGE_SIZE = 100;
+
+  private final int partitionId;
+  private final String pendingUpdateAlias;
+  private final String incidentAlias;
+  private final String listViewAlias;
+  private final String flowNodeAlias;
+  private final String operationAlias;
+  private final OpenSearchAsyncClient client;
+  private final Executor executor;
+  private final Logger logger;
+
+  public OpenSearchIncidentUpdateRepository(
+      final int partitionId,
+      final String pendingUpdateAlias,
+      final String incidentAlias,
+      final String listViewAlias,
+      final String flowNodeAlias,
+      final String operationAlias,
+      final OpenSearchAsyncClient client,
+      final Executor executor,
+      final Logger logger) {
+    this.partitionId = partitionId;
+    this.pendingUpdateAlias = pendingUpdateAlias;
+    this.incidentAlias = incidentAlias;
+    this.listViewAlias = listViewAlias;
+    this.flowNodeAlias = flowNodeAlias;
+    this.operationAlias = operationAlias;
+    this.client = client;
+    this.executor = executor;
+    this.logger = logger;
+  }
+
+  @Override
+  public CompletionStage<PendingIncidentUpdateBatch> getPendingIncidentsBatch(
+      final long fromPosition, final int size) {
+    final var query = createPendingIncidentsBatchQuery(fromPosition);
+    final var request = createPendingIncidentsBatchRequest(size, query);
+
+    try {
+      return client
+          .search(request, PendingIncidentUpdate.class)
+          .thenApplyAsync(this::createPendingIncidentBatch, executor);
+    } catch (final IOException e) {
```

Comment:
:thought_balloon: the client also throws `OpenSearchException` (which is a `RuntimeException`), so it does not have to be catched. I expect that in similar scenarios we are not catching them as well in the ES implementation. 

Is it ok to not fail the future in this case?

Reply:
No, we should catch both. But honestly, looking quickly through their code, this feels more like future proofing or compatibility with the blocking client API.